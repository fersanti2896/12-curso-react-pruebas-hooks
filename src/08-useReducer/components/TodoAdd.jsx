import { useForm } from '../../hooks';

export const TodoAdd = ({ onNewTodo }) => {
    const { onInputChange, description, onResetForm } = useForm({
        description: ''
    });

    const onFormSubmit = ( event ) => {
        event.preventDefault();

        if( description.length <= 1 ) return;

        const newTodo = {
            id: new Date().getTime(),
            done: false,
            description
        }

        onNewTodo( newTodo );
        onResetForm();
    }

    return (
        <>
            <form onSubmit={ (event) => onFormSubmit(event) }>
                <input type='text'
                       className='form-control'
                       placeholder='¿Qué hay que hacer?'
                       name='description'
                       onChange={ onInputChange }
                       value={ description } />

                <button className='btn btn-outline-success mt-2'
                        type='submit' >
                    Agregar
                </button>
            </form>
        </>
    )
}
