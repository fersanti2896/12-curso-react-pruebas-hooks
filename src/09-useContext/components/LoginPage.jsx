import { useContext } from 'react';
import { UserContext } from '../context/UseContext';

export const LoginPage = () => {
    const { user, setUser } = useContext( UserContext );

    return (
        <>
            <h1>LoginPage</h1>
            <hr/>

            <pre>
                { JSON.stringify( user, null, 3 ) }
            </pre>

            <button className='btn btn-outline-success'
                    onClick={ () => setUser({ id: 123, name: 'Fernando', email: 'prueba@gmail.com' }) }>
                Establecer usuario
            </button>
        </>
    )
}
