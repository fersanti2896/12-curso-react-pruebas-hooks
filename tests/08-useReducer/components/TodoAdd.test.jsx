import { render } from "@testing-library/react";
import { TodoAdd } from "../../../src/08-useReducer/components/TodoAdd";

describe('Pruebas en <TodoAdd />', () => { 
    test('Debe hacer match con en el snapshot', () => { 
        const { container } = render( < TodoAdd /> );
        console.log(container)

        expect( container ).toMatchSnapshot();
    });
});