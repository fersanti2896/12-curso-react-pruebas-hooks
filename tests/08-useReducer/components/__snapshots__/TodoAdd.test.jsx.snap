// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> Debe hacer match con en el snapshot 1`] = `
<div>
  <form>
    <input
      class="form-control"
      name="description"
      placeholder="¿Qué hay que hacer?"
      type="text"
      value=""
    />
    <button
      class="btn btn-outline-success mt-2"
      type="submit"
    >
      Agregar
    </button>
  </form>
</div>
`;
